### PR TITLE
Introduce section root client id as block editor setting

### PR DESCRIPTION
**Pull request by draganescu**

### CODE REVIEW

File: packages/block-editor/src/store/actions.js [hunk 1]
```diff
@@ -1461,6 +1461,22 @@ export const __unstableSetEditorMode =
...
+					'' /* rootClientId */,
+					'contentOnly'
+				);
+				dispatch.setBlockEditingMode(
+					sectionRootClientId,
+					'contentOnly'
+				);
+				sectionsClientIds.forEach( ( clientId ) =>
+					dispatch.setBlockEditingMode( clientId, 'default' )
+				);
```
Comment (youknowriad) [inline packages/block-editor/src/store/actions.js:1478]: This logic should ideally move to the `getBlockEditingMode` selector.

File: packages/block-editor/src/store/selectors.js [hunk 2]
```diff
@@ -2814,11 +2814,7 @@ export function __unstableHasActiveBlockOverlayActive( state, clientId ) {
 	const editorMode = __unstableGetEditorMode( state );
 
 	// In zoom-out mode, the block overlay is always active for top level blocks.
-	if (
-		editorMode === 'zoom-out' &&
-		clientId &&
-		! getBlockRootClientId( state, clientId )
-	) {
+	if ( editorMode === 'zoom-out' && clientId ) {
```
Comment (youknowriad) [inline packages/block-editor/src/store/selectors.js:2817]: This should check that the block is either block level or child of sectionRootClientId if defined.

File: packages/editor/src/components/provider/use-block-editor-settings.js [hunk 3]
```diff
@@ -85,17 +88,20 @@ const BLOCK_EDITOR_SETTINGS = [
...
- * @param {string} postId   Editor root level post ID.
+ * @param {Object} settings        EditorProvider settings prop.
+ * @param {string} postType        Editor root level post type.
+ * @param {string} postId          Editor root level post ID.
+ * @param {string} contextPostType The post type of the edited post.
  *
  * @return {Object} Block Editor Settings.
  */
-function useBlockEditorSettings( settings, postType, postId ) {
+function useBlockEditorSettings( settings, postType, postId, contextPostType ) {
```
Comment (youknowriad) [inline packages/editor/src/components/provider/use-block-editor-settings.js:98]: I think ideally the contextPostType should replace the postType argument, and we could pass a "mode" in addition. That way we can fallback to `wp_template` post type if the mode is "template-locked".

File: packages/editor/src/components/provider/use-block-editor-settings.js [hunk 4]
```diff
@@ -230,6 +242,30 @@ function useBlockEditorSettings( settings, postType, postId ) {
 
 	const forceDisableFocusMode = settings.focusMode === false;
 
+	const getSectionRootBlock = useCallback( () => {
+		if ( ! contextPostType ) {
+			return null;
+		}
+		const CONTENT_TYPES = [ 'post', 'page' ];
```
Comment (youknowriad) [inline packages/editor/src/components/provider/use-block-editor-settings.js:249]: These are not the only content types so I'd prefer if remove this. I'd rather do the following algorithm instead:

 1- If mode = "template-locked": look for post-content block 
 2- If mode is not "template-locked', look for the "main" group block
 3- fallback to top level. (undefined section container I guess)

**Review by youknowriad — COMMENTED** [review]
Left some comments, but this approach is cleaner.

**Comment by ajlende** [comment]
@draganescu This approach is better than what we have in #59249, so I've moved the changes over there and started fixing the review comments from here in that PR.

File: packages/block-editor/src/store/actions.js (hunk 1 above)
Comment (ajlende) [reply packages/block-editor/src/store/actions.js:1478]: Fixed in #59249

File: packages/block-editor/src/store/selectors.js (hunk 2 above)
Comment (ajlende) [reply packages/block-editor/src/store/selectors.js:2817]: Fixed in #59249.

**Pull request closed without merging**